### PR TITLE
Creation of hidden photon flux code

### PR DESCRIPTION
**Pull request by tomasoshea**
tomasoshea Large: 907 [](https://github.com/rest-for-physics/axionlib/commits/master) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=rest-for-physics&utm_campaign=badge_info)

Created 2 new files TRestAxionSolarHiddenPhotonFlux.h and TRestAxionSolarHiddenPhotonFlux.cxx to go with the TRestAxionSolarFlux code.

### CODE REVIEW

File: inc/TRestAxionSolarHiddenPhotonFlux.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#define _TRestAxionSolarHiddenPhotonFlux
+
+#include <TRestAxionSolarFlux.h>
+#include <TRestAxionSolarModel.h>
+
+//! A metadata class to load tabulated solar hidden photon fluxes. Mass and coupling set to 1.
+class TRestAxionSolarHiddenPhotonFlux : public TRestAxionSolarFlux {
+   private:
+    /// The hidden photon mass used for calculation
+    double HiddenPhotonMass;
```
Comment (jgalan) [inline inc/TRestAxionSolarHiddenPhotonFlux.h:33]: I think we do not need the mass here? It is the method `GetRandomEnergyAndRadius` where we define the mass. There should be something inside that method that gets the probability depending on the mass.

File: inc/TRestAxionSolarHiddenPhotonFlux.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    /// The value of the kinetic mixing parameter used for calculation
+    double HiddenPhotonKineticMixing;
```
Comment (jgalan) [inline inc/TRestAxionSolarHiddenPhotonFlux.h:36]: This method is no needed. We should use the inherited data members inside `TRestAxionSolarFlux`.

```
std::string fCouplingType = ""; 
Double_t fCouplingStrength = 0;  //<
```

Inside the config file one has to define then:

```
<parameter name="couplingType" value="Chi" />
<parameter name="couplingStrength" value="1.e-10" />
```

File: inc/TRestAxionSolarHiddenPhotonFlux.h (hunk 1 above)
Comment (jgalan) [reply inc/TRestAxionSolarHiddenPhotonFlux.h:33]: Yes, this is possible.

We just need to make sure then that we have this mass inside `TRestAxionSolarFlux` and not within `TRestAxionQCDSolarFlux` or `TRestAxionHiddenPhotonFlux`.

So, we define a `TRestAxionSolarFlux::fMass`.

Then, inside the `TRestAxionGeneratorProcess` we can call `TRestAxionSolarFlux::SetMass(m)` before calling `GetRandomEnergyAndRadius`.

`TRestAxionGeneratorProcess` will only have access to those methods available to `TRestAxionSolarFlux`.

Comment (jgalan) [reply inc/TRestAxionSolarHiddenPhotonFlux.h:33]: Then, inside `TRestAxionSolarFlux` we should add:

```
Double_t fMass; //!

Double_t GetMass() { return fMass; }
void SetMass(const Double_t &m ){ fMass = m; }
```

Comment (tomasoshea) [reply inc/TRestAxionSolarHiddenPhotonFlux.h:33]: Ok, then would we remove the `double_t mass` from the other functions and just have them use `fMass`? For example in `GetTotalFlux(Double_t mass = 0)`

**Pull request merged**